pkg/proc/native/threads_darwin.go

```diff
@@ -1,4 +1,4 @@
-//+build darwin,macnative
+//+build !arm64,macnative
```

Comment:
The file name should be changed to threads_darwin_amd64.go and the build constraint should be `darwin,amd64,macnative`.

Reply:
I think you were looking at an old version of the code. I have extracted the common parts of the Darwin threads into threads_darwin.go, the specifics go into the _arm64 and _amd64.go files. Since the only specific thing is the thread_state struct I was actually thinking about a typedef to avoid having those platform specific files alltogether.